web/src/components/molecules/Content/ContentForm/index.tsx

```diff
@@ -0,0 +1,181 @@
+import styled from "@emotion/styled";
+import Upload from "antd/lib/upload/Upload";
+import { useCallback } from "react";
+
+import Button from "@reearth-cms/components/atoms/Button";
+import Form from "@reearth-cms/components/atoms/Form";
+import Icon from "@reearth-cms/components/atoms/Icon";
+import Input from "@reearth-cms/components/atoms/Input";
+import InputNumber from "@reearth-cms/components/atoms/InputNumber";
+import PageHeader from "@reearth-cms/components/atoms/PageHeader";
+import Select from "@reearth-cms/components/atoms/Select";
+import TextArea from "@reearth-cms/components/atoms/TextArea";
+import { ItemField } from "@reearth-cms/components/molecules/Content/types";
+import { FieldType, Model } from "@reearth-cms/components/molecules/Schema/types";
+import { useT } from "@reearth-cms/i18n";
+import { validateURL } from "@reearth-cms/utils/regex";
+
+export interface Props {
+  itemId?: string;
+  initialFormValues: any;
+  defaultFormValues: any;
+  model?: Model;
+  onItemCreate: (data: { schemaID: string; fields: ItemField[] }) => Promise<void>;
+  onItemUpdate: (data: { itemID: string; fields: ItemField[] }) => Promise<void>;
+  onBack: () => void;
+}
+
+const ContentForm: React.FC<Props> = ({
+  itemId: itemID,
+  model,
+  initialFormValues,
+  defaultFormValues,
+  onItemCreate,
+  onItemUpdate,
+  onBack,
+}) => {
+  const t = useT();
+  const { Option } = Select;
+
+  const [form] = Form.useForm();
+
+  const handleSubmit = useCallback(async () => {
+    try {
+      const values = await form.validateFields();
+      const fields: { schemaFieldID: string; type: FieldType; value: string }[] = [];
+      for (const [key, value] of Object.entries(values)) {
+        fields.push({
+          value: (value || "") as string,
+          schemaFieldID: key,
+          type: model?.schema.fields.find(field => field.id === key)?.type as FieldType,
+        });
+      }
+      if (!itemID) await onItemCreate?.({ schemaID: model?.schema.id as string, fields });
+      else await onItemUpdate?.({ itemID: itemID as string, fields });
+    } catch (info) {
+      console.log("Validate Failed:", info);
+    }
+  }, [form, model?.schema.fields, model?.schema.id, itemID, onItemCreate, onItemUpdate]);
+
+  return (
+    <Form
+      form={form}
+      layout="vertical"
+      initialValues={itemID ? initialFormValues : defaultFormValues}>
```

Comment:
This has the potential to be undefined (because you allow initialFormValues to be `any`), which since you have default values I'm guessing you don't want.

Reply:
@KaWaite , it is typed as `any` from the backend, in frontend it is looping through the backend data keys, so the only way to make it have an `undefined` value, is to have the backend return undefined for it.
Also if it is `undefined` or `null`, there won't be a value in form and that's the correct behavior.

Reply:
Ok that much makes sense, but the logic can be cleared up a little more.
So, you have an array of initial values, and an array of default values. 

Some thoughts:

When you generate the initial values, you are already filtering with `itemId`, so if there is no itemId, initialFormValues will become undefined. So `initialFormValues ?? defaultFormValues` would be enough in this situation.

Then again, when an item is created, is it fully blank, and only user changed fields will have values, just like reearth-web? If so, you'll need to combine the initial and default values (looping over the schema).

Reply:
@KaWaite , no initial value is only showing when creating new content.
initialFormValues returns an empty object, and I need to check if it's updating or creating, for that purpose I'm using: 
`itemID ? initialFormValues : defaultFormValues`